milli/src/update/delete_documents.rs

```diff
@@ -380,6 +380,23 @@ impl<'t, 'u, 'i> DeleteDocuments<'t, 'u, 'i> {
 
         drop(iter);
 
+        if let Some(mut rtree) = self.index.geo_rtree(self.wtxn)? {
+            let mut geo_faceted_doc_ids = self.index.geo_faceted_documents_ids(self.wtxn)?;
+
+            let points_to_remove: Vec<_> = rtree
+                .iter()
+                .filter(|&point| self.documents_ids.contains(point.data))
+                .cloned()
+                .collect();
+            points_to_remove.iter().for_each(|point| {
+                rtree.remove(&point);
+                geo_faceted_doc_ids.remove(point.data);
+            });
```

Comment:
```suggestion
            let (points_to_remove, docids_to_remove): (Vec<_>, RoaringBitmap) = rtree
                .iter()
                .filter(|&point| self.documents_ids.contains(point.data))
                .cloned()
                .map(|point| (point, point.data))
                .collect();
            points_to_remove.iter().for_each(|point| {
                rtree.remove(&point);
            });
            geo_faceted_doc_ids -= docids_to_remove;
```

Reply:
Funny, rust was not understanding this collect but `unzip` worked